src/main/java/seedu/address/model/person/Class.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.model.person;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+/**
+ * Represents a Person's class date in the address book.
+ * Guarantees: immutable.
+ */
+public class Class {
+    public static final String MESSAGE_CONSTRAINTS = "Class can take any string"
+            + " in the format of 'yyyy-MM-dd 0000-2300'";
+    public static final String INVALID_DATETIME_ERROR_MESSAGE = "Date should be a valid date";
+    public static final String INVALID_TIME_ERROR_MESSAGE = "Time should be in the range of 0000 - 2359";
+    public static final String VALIDATION_DATETIME_REGEX = "[0-9]{4}[-][0-9]{2}[-][0-9]{2}";
+    public static final String VALIDATION_TIME_REGEX = "[0-9]{4}";
+    public static final String VALIDATION_CLASS_REGEX = VALIDATION_DATETIME_REGEX
+            + "[ ]" + VALIDATION_TIME_REGEX + "[-]" + VALIDATION_TIME_REGEX;
+    public final LocalDate date;
+    public final LocalTime startTime;
+    public final LocalTime endTime;
+
+
+    /**
+     * Constructs a {@code Class}.
+     *
+     * @param date
+     * @param startTime
+     * @param endTime
+     */
+    public Class(LocalDate date, LocalTime startTime, LocalTime endTime) {
+        this.date = date;
+        this.startTime = startTime;
+        this.endTime = endTime;
+    }
+
+    /**
+     * Validates whether classDatetime is valid.
+     * @param classDatetime the string to be validated.
+     * @return true if a given string fits the format of 'yyyy-MM-d 0000-2300'
+     */
+    public static boolean isValidClassString(String classDatetime) {
+        if (!classDatetime.matches(VALIDATION_CLASS_REGEX)) {
+            return false;
+        }
+
+        String datetimeStr = classDatetime.substring(0, 10);
+        String startTimeStr = classDatetime.substring(11, 15);
+        String endTimeStr = classDatetime.substring(16);
+
+        return isValidDatetimeString(datetimeStr) && isValidTimeString(startTimeStr) && isValidTimeString(endTimeStr);
+    }
+
+    /**
+     * Helper method to validate {@code date}
```

Comment:
1. Following the code quality convention, maybe this line can be 
`Validates {@code date}`
with an action verb as the first word?
2. Also, it would be nice to include a return tag since a boolean is returned!
3. Seems like the first line ends with a full stop for other javadoc comments?